test/integration/newapp_test.go

```diff
@@ -1711,7 +1712,7 @@ func TestNewAppSourceAuthRequired(t *testing.T) {
 	}
 }
 
-func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) string {
+func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) (string, func()) {
 	// Create test directories
 	testDir, err := ioutil.TempDir("", "gitauth")
```

Comment:
> What you're talking looks like your personal opinion to me.

What I am describing is how every single one of our test suites runs ... run a bunch of them and look at the tree under `/tmp/openshift`.

> It would be useful if you give me links with explanation of how we chose it should be.

I don't think we have a design document for something as simple as temporary directory file structure ... but you can look at the temporary directory setup [function](https://github.com/openshift/origin/blob/master/hack/lib/util/environment.sh#L115-L161) [called](https://github.com/openshift/origin/blob/master/hack/test-cmd.sh#L108) [in](https://github.com/openshift/origin/blob/master/hack/test-end-to-end-docker.sh#L13) [all](https://github.com/openshift/origin/blob/master/hack/test-end-to-end.sh#L46) [of](https://github.com/openshift/origin/blob/master/hack/test-go.sh#L42) [our](https://github.com/openshift/origin/blob/master/hack/test-integration.sh#L9) [test](https://github.com/openshift/origin/blob/master/test/extended/setup.sh#L36) [scripts](https://github.com/openshift/origin/blob/master/test/extended/ldap_groups.sh#L26) if you don't believe me... 

> Also I see that integration tests create files not how you expect that they should do.

And, again, I tell you this is incorrect behavior and a bug. I'm not asking you to fix all of these, but at least while you're changing this bit of code, make it place them in the right place.